### PR TITLE
Issue1340 weather data

### LINKED ISSUE #1340: change radiation of weather data to exactly zero rather than 1E-4
The weather data reader always produces slightly positive radiation values. Looking at `WeatherData.Examples.ReaderTMY3`, one can see that if the radiation is set to 0, then `HDirNor`, `HGloHor` and `HDifHor` are 0.0001 W/m2. We set them to a slightly positive value because time interpolation of weather data can lead to slightly negative values, and the `smoothMax` was used to avoid events. In the meantime, MSL ensures that `max` does not generate events. This issue is to test if we could set these values to exactly 0 and avoid using the `smoothMax` here.

This will need to be tested on larger models, e.g., as part of the Buildings library.

The reason is that non-zero values do not allow achieving steady-state in a building model, which a user tried to accomplish as part of controls design.

### PR DESCRIPTION
**Pull request by mwetter**
This closes #1340

### CODE REVIEW

**Comment by mwetter** [comment]
@Mathadon : This is ready for review, see also comments and tests on Buildings library that are reported on https://github.com/ibpsa/modelica-ibpsa/issues/1340

File: IBPSA/BoundaryConditions/WeatherData/BaseClasses/PartialLimiter.mo [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+annotation (
+defaultComponentName="lim",
+Documentation(info="<html>
+<p>
+Block that computes <i>y_internal=min(uMax, max(uMin, u)</i>,
+where <code>y_internal</code> is a protected connector.
+</p>
+<p>
+This block is used because interpolation of weather data can lead to
```
Comment (Mathadon) [inline IBPSA/BoundaryConditions/WeatherData/BaseClasses/PartialLimiter.mo:20]: Thx for documenting this.

File: IBPSA/Resources/ReferenceResults/Dymola/IBPSA_BoundaryConditions_WeatherData_Examples_ReaderTMY3.txt [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
+weaDat.weaBus.HGloHor=[0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00, 0e+00]
```
Comment (Mathadon) [inline IBPSA/Resources/ReferenceResults/Dymola/IBPSA_BoundaryConditions_WeatherData_Examples_ReaderTMY3.txt:11]: pretty.

File: IBPSA/BoundaryConditions/WeatherData/BaseClasses/LimiterHorizontalInfraredIrradiation.mo [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+block LimiterHorizontalInfraredIrradiation
+  "Block that limits the horizontal infrared irradiation"
+  extends PartialLimiterMin;
+
+  Modelica.Blocks.Interfaces.RealOutput HHorIR(
+    final unit="W/m2") "Horizontal infrared irradiation"
+    annotation (Placement(transformation(extent={{100,-10},{120,10}})));
+
+equation
+  HHorIR = max(0, u);
```
[the remark this answers is not in the document]
Comment (mwetter) [reply IBPSA/BoundaryConditions/WeatherData/BaseClasses/LimiterHorizontalInfraredIrradiation.mo:11]: @Mathadon : Is there indeed a difference for the symbolic processing between these statements?:
```
Real y = max(u,0);
```
and
```
Real y;
equation
y = max(u, 0);
```
I am not aware of it.
At least Dymola solves analytically
```
model Unnamed
  Real y=2*u;
  Real u=1+y;
end Unnamed;
```

**Comment by mwetter** [comment]
This is ready to merge when all tests pass.

File: IBPSA/BoundaryConditions/WeatherData/BaseClasses/LimiterHorizontalInfraredIrradiation.mo (hunk 3 above)
[the remark this answers is not in the document]
Comment (Mathadon) [reply IBPSA/BoundaryConditions/WeatherData/BaseClasses/LimiterHorizontalInfraredIrradiation.mo]: As far as I could tell from the code, in JModelica the first formulation is treated as a 'binding equation'. For these equations, JModelica knows up front that `y` is solved from `max(u,0)`. For the second formulation, the equation `y = max(u, 0)` could either be solved for `y`, or for `u`. The translator doesn't know that up front. After alias elimination etc, the BLT transformation is done, where all remaining equations (excluding binding equations) are processed to figure out what equation is solved for what variable. This algorithm takes time, so the fewer equations that are part of that set, the better, I figured.

I don't know how Dymola's processing works of course.

**Pull request merged**